packages/react-composites/tests/browser/chat/ChatComposite.test.ts

```diff
@@ -63,21 +70,56 @@ test.describe('Chat Composite E2E Tests', () => {
     await browser.close();
   });
 
-  test('composite loads correctly with participant list', async () => {
-    await pages[0].bringToFront();
-    await pages[0].waitForSelector(dataUiId(IDS.sendboxTextfield));
-    await pages[0].waitForSelector(dataUiId(IDS.participantList));
-    stubTimestamps(pages[0]);
-    expect(await pages[0].screenshot()).toMatchSnapshot('1-chat-screen.png');
+  test('composite pages load completely', async () => {
+    for (const idx in pages) {
+      await compositeLoaded(pages[idx]);
+      expect(await pages[idx].screenshot()).toMatchSnapshot(`1-page-${idx}-chat-screen.png`);
+    }
   });
 
   test('can send message', async () => {
+    const page = pages[0];
+    await page.bringToFront();
+    await page.type(dataUiId(IDS.sendboxTextfield), 'How the turn tables');
+    await page.keyboard.press('Enter');
+    await page.waitForSelector(`[data-ui-status="delivered"]`);
+    stubTimestamps(page);
+    expect(await page.screenshot()).toMatchSnapshot('2-send-message.png');
+  });
+
+  test('can receive message', async () => {
+    const page = pages[1];
+    await page.bringToFront();
+    await page.waitForSelector(`[data-ui-status="delivered"]`);
+    stubTimestamps(page);
+    expect(await page.screenshot()).toMatchSnapshot('3-receive-message.png');
+  });
+
+  test('read message has a viewed status', async () => {
+    const page = pages[0];
+    await page.bringToFront();
+    await page.waitForSelector(`[data-ui-status="seen"]`);
+    stubTimestamps(page);
+    expect(await page.screenshot()).toMatchSnapshot('4-read-message-status.png');
+  });
+
+  test('can view typing indicator', async () => {
+    const page = pages[1];
+    await page.bringToFront();
+    await page.type(dataUiId(IDS.sendboxTextfield), 'I am not superstitious. Just a little stitious.');
     await pages[0].bringToFront();
-    await pages[0].waitForSelector(dataUiId(IDS.sendboxTextfield));
-    await pages[0].type(dataUiId(IDS.sendboxTextfield), 'How the turn tables');
-    await pages[0].keyboard.press('Enter');
-    await pages[0].waitForSelector(`[data-ui-status="delivered"]`);
-    stubTimestamps(pages[0]);
-    expect(await pages[0].screenshot()).toMatchSnapshot('2-send-message.png');
+    await pages[0].waitForSelector(dataUiId(IDS.typingIndicator));
+    const el = await pages[0].$(dataUiId(IDS.typingIndicator));
+    expect(await el?.innerHTML()).toContain(participants[1]);
+    expect(await pages[0].screenshot()).toMatchSnapshot('5-typing-indicator.png');
+  });
+
+  test('typing indicator disappears after 12 seconds', async () => {
+    const page = pages[0];
+    await page.bringToFront();
+    await page.waitForTimeout(12000);
+    const el = await page.$(dataUiId(IDS.typingIndicator));
+    expect(await el?.innerHTML()).toBeFalsy();
+    expect(await page.screenshot()).toMatchSnapshot('6-typing-indicator-disappears.png');
```

Comment:
If this fails, how do we view the diff you showed before? If its via running locally only, is there a command I can run to get view the diff for a failed/named test?

Reply:
Likewise for reviewers is there a way they could see the diff also ?

Reply:
If you run this locally and the outcome doesn't match. A new dirrectory `test-result` will be created inside `react-composite` with the old, new and diff images